modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java

```diff
@@ -159,7 +168,12 @@ public void stop() {
                 }
 
                 cluster.shutdown();
-                cluster.onShutdown().block();
+
+                try {
+                    shutdownFuture.get(10, TimeUnit.SECONDS);
+                } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                    throw new IgniteInternalException("Unable to stop the ClusterService", e);
```

Comment:
Shouldn't we restore interruption status if we caught `InterruptedException` here using `Thread.interrupt()`?